lib/private/Template/JSResourceLocator.php

```
@@ -116,14 +131,30 @@ public function doFind($script) {
 	public function doFindTheme($script) {
 	}
 
+	/**
+	 * Try to find ES6 script file (`.mjs`) with fallback to plain javascript (`.js`)
+	 * @see appendIfExist()
+	 */
+	protected function appendScriptIfExist($root, $file, $webRoot = null) {
+		if (!$this->appendIfExist($root, $file . '.mjs', $webRoot)) {
```

because of the changes above that prevent throwing AppPathNotFoundException it could happen that `$root` and `$webRoot` are null

to preserve the original behavior you'd likely need to throw that exception here after checking

---

This is the original behavior, as previously `\OC_App::getAppPath` was used which did not trow but return `false` if not found.
So this behavior is kept by assigning `false` to `$app_path` and `$app_url` and just ignore `AppPathNotFoundException` above (if it is raised, the value would stay `false`).

If `$app_path` is `false` then `appendIfExist` does skip it and if both are `false, then this is handled by the line 115 (next code below the `if(strpos`  block)
```php
if ($app_path === false && $app_url === false) {
// ...
```

But I agree that this is not that obvious from the code without digging into it. I tried to make this more clear in a separate commit, but I am not sure if it should be part of this pull request: https://github.com/susnux/nextcloud/commit/4d00f5624e348734b25d657cd9715a9b97a0a725